src/Compilers/CSharp/Test/Semantic/Semantics/RawInterpolationTests.cs

```diff
@@ -0,0 +1,1276 @@
+        CompileAndVerify(source, expectedOutput: expectedOutput);
+    }
+
+    [Fact]
+    public void OneLiteral()
+    {
+        string source =
+@"using System;
+class Program
+{
+    static void Main(string[] args)
+    {
+        Console.WriteLine( $""""""Hello"""""" );
+    }
+}";
+        string expectedOutput = @"Hello";
+        var verifier = CompileAndVerify(source, expectedOutput: expectedOutput);
+        verifier.VerifyIL("Program.Main", @"
+{
+  // Code size       11 (0xb)
+  .maxstack  1
+  IL_0000:  ldstr      ""Hello""
+  IL_0005:  call       ""void System.Console.WriteLine(string)""
+  IL_000a:  ret
+}
+");
+    }
+
+    [Fact]
+    public void OneInsert()
+    {
+        string source =
+@"using System;
+class Program
+{
+    static void Main(string[] args)
+    {
+        var hello = $""""""Hello"""""";
+        Console.WriteLine( $""""""{hello}"""""" );
+    }
+}";
+        string expectedOutput = @"Hello";
+        var verifier = CompileAndVerify(source, expectedOutput: expectedOutput);
+        verifier.VerifyIL("Program.Main", @"
+{
+  // Code size       20 (0x14)
+  .maxstack  2
+  IL_0000:  ldstr      ""Hello""
+  IL_0005:  dup
+  IL_0006:  brtrue.s   IL_000e
+  IL_0008:  pop
+  IL_0009:  ldstr      """"
+  IL_000e:  call       ""void System.Console.WriteLine(string)""
+  IL_0013:  ret
+}
+");
+    }
+
+    [Fact]
+    public void TwoInserts()
+    {
+        string source =
+@"using System;
+class Program
+{
+    static void Main(string[] args)
+    {
+        var hello = $""""""Hello"""""";
+        var world = $""""""world"""""" ;
+        Console.WriteLine( $""""""{hello}, { world }."""""" );
+    }
+}";
+        string expectedOutput = @"Hello, world.";
+        CompileAndVerify(source, expectedOutput: expectedOutput);
+    }
+
+    [Fact]
+    public void TwoInserts02()
+    {
+        string source =
+@"using System;
+class Program
+{
+    static void Main(string[] args)
+    {
+        var hello = ""Hello"";
+        var world = ""world"";
+        Console.WriteLine( $""""""
+                            {
+                                    hello
+                            },
+{
+                            world }.
+                            """""" );
+    }
+}";
+        string expectedOutput = @"Hello,
+world.";
+        CompileAndVerify(source, expectedOutput: expectedOutput);
+    }
+
+    [Fact, WorkItem(306, "https://github.com/dotnet/roslyn/issues/306"), WorkItem(308, "https://github.com/dotnet/roslyn/issues/308")]
+    public void DynamicInterpolation()
+    {
+        string source =
+@"using System;
+using System.Linq.Expressions;
+class Program
+{
+    static void Main(string[] args)
+    {
+        dynamic nil = null;
+        dynamic a = new string[] {""""""Hello"""""", """"""world""""""};
+        Console.WriteLine($""""""<{nil}>"""""");
+        Console.WriteLine($""""""<{a}>"""""");
+    }
+    Expression<Func<string>> M(dynamic d) {
```

Comment:
These are just copied from the string interp tests right? Is this expression actually needed for the test correctness?

Reply:
correct.